OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseSubSectionAdapter.kt

```diff
@@ -107,20 +110,20 @@ class CourseSubSectionAdapter(
 
             setBulkDownloadView(component)
             binding.root.setOnClickListener {
-                itemClickListener.onItemClick(binding.root, position)
+                itemClickListener.onSubSectionItemClick(binding.root, position)
             }
 
             binding.root.setOnLongClickListener {
-                itemClickListener.onItemLongClick(binding.root, position)
+                itemClickListener.onSubSectionLongClick(binding.root, position)
                 return@setOnLongClickListener true
             }
         }
 
         private fun setBulkDownloadView(component: CourseComponent) {
```

Comment:
Some stuff related to this function: 
- IMO, we can remove all three of the comments in this function (2 are vague and 1 is over-explaining)
- Move the arguments of downloadedCount to new liens as well
- And a few more things 😂 

In short something like this: 
```
        private fun setBulkDownloadView(component: CourseComponent) {
            val totalDownloadableVideos: Int = component.downloadableVideosCount
            if (totalDownloadableVideos == 0) {
                binding.bulkDownloadLayout.bulkDownloadLayout.setVisibility(false)
                return
            }

            val path = component.path
            val chapterId = path.get(1)?.displayName ?: ""
            val sequentialId = path.get(2)?.displayName ?: ""
            val downloadedCount: Int = dbStore.getDownloadedVideosCountForSection(
                component.courseId,
                chapterId,
                sequentialId,
                null
            )
            val downloadingVideosCount = dbStore.getDownloadingVideosCountForSection(
                component.courseId,
                chapterId,
                sequentialId,
                null
            )

            binding.bulkDownloadLayout.root.setVisibility(true)
            binding.bulkDownloadLayout.bulkDownload.setVisibility(true)
            binding.bulkDownloadLayout.noOfVideos.apply {
                setVisibility(true)
                text = totalDownloadableVideos.toString()

                if (downloadedCount == totalDownloadableVideos) {
                    setRowStateOnDownload(DownloadedState.DOWNLOADED)
                } else if (downloadingVideosCount + downloadedCount == totalDownloadableVideos) {
                    setRowStateOnDownload(DownloadedState.DOWNLOADING) {
                        itemClickListener.viewDownloadsStatus()
                    }
                } else {
                    val downloadableVideos = component.getVideos(true)
                        .filterIsInstance<VideoBlockModel>()
                        .onEach { videoBlockModel ->
                            videoBlockModel.downloadUrl =
                                VideoUtil.getPreferredVideoUrlForDownloading(videoBlockModel.data)
                        }
                    setRowStateOnDownload(DownloadedState.ONLINE) {
                        itemClickListener.download(downloadableVideos = downloadableVideos)
                    }
                }
            }
        }

Reply:
There is some space extra new lines above. Please adjust them as well.

Reply:
formated.